### ISSUE TITLE
Please comment the code!

### ISSUE THREAD

**Issue by 10maurycy10** [mxgmn/WaveFunctionCollapse#78]
Comments make reading and debugging the code much easier.

They are especially necessary if the code has one letter variable names!

**Comment by mxgmn** [comment]
I agree. Will add a few comments when I have time. Right now I can say that in `SimpleTiledModel.cs` `a` is a 90 degrees rotation, and `b` is tile reflection.

**Comment by segrallx** [comment]
hello，what's the meaning of symmetry?  
cardinality  ?
func a b ? 

need some comments. thanks..
or is any place we can talk about wfc . maybe anybody need some help or communicate.

**Comment by appgurueu** [comment]
> hello，what's the meaning of symmetry?

Google it.

> cardinality ?

Google it.

Comments are not the place for defining basic mathematical terms.

**Comment by segrallx** [comment]
thank you , i still consued about the meaning of  X ,F,  L ,T
could you tell me what symmetry they are representing?

**Comment by mxgmn** [comment]
X, F, L, T are types of tile symmetries. X means a fully symmetric tile: you can rotate it 90, 180, 270 degrees or reflect it, and it would still look the same (like letter X). L can be reflected along the line y=x. F has no nontrivial symmetries.

**Comment by ldo** [comment]
Cardinality: how many orientations (excluding reflection) of a tile with this symmetry can be distinguished.

`Func<int, int> a, b`: I call `a` the `rotate_func` — the function that transforms the index of a tile orientation to the one for the next 90° rotation anticlockwise — and `b` the `reflect_func` — the function that transforms the index of a tile orientation to the one for the reflected version.

Also worth pointing out that all the quantum mumbo-jumbo is a complete red herring. If you want a physical analogy, use the term “phase change”.

**Comment by Gouvernathor** [comment]
I suggest the following renames, mostly in Model for now (possibly more coming) :
- wave -> outTiles
- T -> nSourceTiles
- MX -> nOutTilesX
- MY -> nOutTilesY
- N -> patternSizeInPixels

The case is following JS conventions, so you can ignore it in favor of C# conventions (which are beyond my comprehension), but otherwise that's what I recommend.

Update : actually I'm not sure that N is actually the size of a pattern tile in pixels. Which goes back to the original request : *please comments*
